forc-plugins/forc-doc/src/render.rs

```diff
@@ -300,6 +156,186 @@ impl RenderedDocumentation {
+            .1
+            .to_vec();
+        match module_map.get_mut(&module_prefixes) {
+            Some(doc_links) => match doc_links.get_mut(&BlockTitle::Modules) {
+                Some(links) => {
+                    if !links.contains(&module_link) {
+                        links.push(module_link)
+                    }
+                }
+                None => {
+                    doc_links.insert(BlockTitle::Modules, vec![module_link]);
+                }
+            },
+            None => {
+                let mut doc_links: BTreeMap<BlockTitle, Vec<DocLink>> = BTreeMap::new();
+                doc_links.insert(BlockTitle::Modules, vec![module_link]);
+                module_map.insert(module_prefixes.clone(), doc_links);
+            }
+        }
+        module_clone.module_prefixes.pop();
+    }
+}
+fn populate_all_doc(doc: &Document, all_docs: &mut DocLinks) {
+    match doc.item_body.ty_decl {
+        StructDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Structs) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs.links.insert(BlockTitle::Structs, vec![doc.link()]);
+            }
+        },
+        EnumDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Enums) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs.links.insert(BlockTitle::Enums, vec![doc.link()]);
+            }
+        },
+        TraitDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Traits) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs.links.insert(BlockTitle::Traits, vec![doc.link()]);
+            }
+        },
+        AbiDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Abi) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs.links.insert(BlockTitle::Abi, vec![doc.link()]);
+            }
+        },
+        StorageDecl { .. } => match all_docs.links.get_mut(&BlockTitle::ContractStorage) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs
+                    .links
+                    .insert(BlockTitle::ContractStorage, vec![doc.link()]);
+            }
+        },
+        FunctionDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Functions) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs
+                    .links
+                    .insert(BlockTitle::Functions, vec![doc.link()]);
+            }
+        },
+        ConstantDecl { .. } => match all_docs.links.get_mut(&BlockTitle::Constants) {
+            Some(links) => links.push(doc.link()),
+            None => {
+                all_docs
+                    .links
+                    .insert(BlockTitle::Constants, vec![doc.link()]);
+            }
+        },
+        _ => {} // TODO: ImplTraitDeclaration
+    }
+}
```

Comment:
I think this can be shortened using the helper function from my previous comment:

```
fn populate_all_doc(doc: &Document, all_docs: &mut DocLinks) {
    populate_doc_links_for_decl(doc, &mut all_docs.links);
}
```